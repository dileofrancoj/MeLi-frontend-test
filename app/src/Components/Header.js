import React from "react";
import { Form, Row, Col, InputGroup, Button } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import { Link } from "react-router-dom";

const heightCol = {
  height: "40px",
};

/* Search function */
const Header = ({ handleSearch }) => {
  const history = useHistory();
  console.log(history);
  const search = (e) => {
    const [{ value }] = e.target;
    e.preventDefault();
    // prevenir busqueda vac√≠os
    if (value.trim() !== "") {
      handleSearch(value);
      history.push(`/items?search=${value}`);
    }
  };

  return (
    <header className="app-header">
      <Row
        className="align-items-center searchBox m-0"
        style={{ textAlign: "right" }}
      >
        <Col md={4} style={heightCol}>
          <img
            alt="logo"
            src="https://images-prod-meli.s3-sa-east-1.amazonaws.com/Logo_ML.png"
            onClick={() => history.goBack()}
          />
        </Col>
        <Col md={4} style={heightCol}>
          <Form onSubmit={search}>
            <Form.Group controlId="formBasicEmail">
              <InputGroup>
                <input
                  name="search"
                  className="input-search form-control"
                  placeholder="Nunca dejes de buscar"
                  id="search"
                />

                <Button type="submit" className="btn-search">
                  <img
                    src="https://images-prod-meli.s3-sa-east-1.amazonaws.com/ic_Search.png"
                    alt="search icon"
                  />
                </Button>
              </InputGroup>
            </Form.Group>
          </Form>
        </Col>
      </Row>
    </header>
  );
};

export default Header;
