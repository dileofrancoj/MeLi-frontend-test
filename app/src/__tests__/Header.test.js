import React from "react";
import { shallow } from "enzyme";

import Header from "../Components/Header";

describe("App", () => {
  test("should render Header component", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
  test("should simulate search without hooks", () => {
    const component = shallow(<Header />);
    const searchBox = component.find("#seach");
    searchBox.value = "macbook";
    component.find("button").at(0).simulate("click");
    expect(searchBox.value).toBe("macbook");
  });
});
