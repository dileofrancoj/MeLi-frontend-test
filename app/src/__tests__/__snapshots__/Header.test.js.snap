// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Header component 1`] = `
<header
  className="app-header"
>
  <[object Object]
    className="align-items-center searchBox m-0"
    noGutters={false}
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    <[object Object]
      md={4}
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <[object Object]
        to="/products"
      >
        <img
          alt="logo"
          src="https://images-prod-meli.s3-sa-east-1.amazonaws.com/Logo_ML.png"
        />
      </[object Object]>
    </[object Object]>
    <[object Object]
      md={4}
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <[object Object]
        inline={false}
        onSubmit={[Function]}
      >
        <[object Object]
          controlId="formBasicEmail"
        >
          <[object Object]>
            <input
              className="input-search form-control"
              id="search"
              name="search"
              placeholder="Nunca dejes de buscar"
            />
            <[object Object]
              active={false}
              className="btn-search"
              disabled={false}
              type="submit"
              variant="primary"
            >
              <img
                alt="search icon"
                src="https://images-prod-meli.s3-sa-east-1.amazonaws.com/ic_Search.png"
              />
            </[object Object]>
          </[object Object]>
        </[object Object]>
      </[object Object]>
    </[object Object]>
  </[object Object]>
</header>
`;

exports[`Header should render Header component  1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <Header />
</Router>
`;
